provider-directory/src/main/java/gov/va/api/health/providerdirectory/service/controller/practitionerrole/PractitionerRoleController.java

```diff
@@ -86,20 +67,46 @@
             PractitionerRole.Bundle::new));
   }
 
-  @SneakyThrows
-  private ProviderContacts ppmsProviderContact(String id) {
-    return ppmsClient.providerContactsForId(id);
-  }
-
-  @SneakyThrows
-  private PpmsProviderSpecialtiesResponse ppmsProviderSpecialty(String id) {
-    return ppmsClient.providerSpecialtySearch(id);
-  }
-
   /** Read by identifier. */
   @GetMapping(value = {"/{publicId}"})
-  public PractitionerRole read(@PathVariable("publicId") String publicId) {
-    throw new UnsupportedOperationException();
+  public PractitionerRole readByIdentifier(@PathVariable("publicId") String publicId) {
+    return transformer.apply(search(Parameters.forIdentity((publicId))));
+  }
+
+  private PractitionerRoleWrapper search(MultiValueMap<String, String> parameters) {
+    ProviderResponse providerResponse;
+    if (parameters.containsKey("identifier")) {
+      String identifier = parameters.getFirst("identifier");
+      providerResponse = ppmsClient.providersForId(identifier);
+    } else if (parameters.containsKey("name")) {
+      String name = parameters.getFirst("name");
+      providerResponse = ppmsClient.providersForName(name);
+    } else if (parameters.containsKey("family") && parameters.containsKey("given")) {
+      String familyName = parameters.getFirst("family");
+      String givenName = parameters.getFirst("given");
+      providerResponse = ppmsClient.providersForName(familyName);
+      List<ProviderResponse.Value> providerResponseFiltered = new ArrayList<>();
+      for (int i = 0; i < providerResponse.value().size(); i++) {
```

Comment:
A for-each loop is a little cleaner here: `for (ProviderResponse.Value val : providerResponse.value())`